lib/Dialect/OM/OMOps.cpp

```diff
@@ -145,6 +145,62 @@ circt::om::ObjectOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ObjectFieldOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult
+circt::om::ObjectFieldOp::verifySymbolUses(SymbolTableCollection &symbolTable) {
+  // Get the containing ModuleOp.
+  auto moduleOp = getOperation()->getParentOfType<ModuleOp>();
+
+  // Get the ObjectInstOp and the ClassOp it is an instance of.
+  ObjectOp objectInst = getObject().getDefiningOp<ObjectOp>();
+  ClassOp classDef = cast<ClassOp>(
+      symbolTable.lookupSymbolIn(moduleOp, objectInst.getClassNameAttr()));
+
+  // Traverse the field path, verifying each field exists.
+  Value finalField;
+  auto fields = SmallVector<FlatSymbolRefAttr>(
+      getFieldPath().getAsRange<FlatSymbolRefAttr>());
+  for (size_t i = 0, e = fields.size(); i < e; ++i) {
+    // Verify the field exists on the ClassOp.
+    auto field = fields[i];
+    ClassFieldOp fieldDef =
+        cast_or_null<ClassFieldOp>(symbolTable.lookupSymbolIn(classDef, field));
+    if (!fieldDef) {
+      auto error = emitOpError("referenced non-existant field ") << field;
+      error.attachNote(classDef.getLoc()) << "class defined here";
+      return error;
+    }
+
+    // If there are more fields, verify the current field is of ClassType, and
+    // look up the ClassOp for that field.
+    if (i < e - 1) {
+      auto classType = fieldDef.getValue().getType().dyn_cast<ClassType>();
+      if (!classType)
+        return emitOpError("nested field access into ")
+               << field << " requires a ClassType, but found "
+               << fieldDef.getValue().getType();
+
+      // The nested ClassOp must exist, since a field with ClassType must be
+      // an ObjectInstOp, which already verifies the class exists.
+      classDef = cast<ClassOp>(
+          symbolTable.lookupSymbolIn(moduleOp, classType.getClassName()));
+    }
+
+    finalField = fieldDef.getValue();
```

Comment:
nit:
```suggestion
    } else {
      finalField = fieldDef.getValue();
    }
```

Reply:
Thanks for pointing it out, this is confusing to update `finalField` on every iteration. Rather than using `else` I opted to add a `continue` at the end of the `if` block with a comment.